grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/HttpStreamDeframer.java

```diff
@@ -60,6 +64,7 @@ public HttpStreamDeframer(
         this.decompressorRegistry = requireNonNull(decompressorRegistry, "decompressorRegistry");
         this.transportStatusListener = requireNonNull(transportStatusListener, "transportStatusListener");
         this.statusFunction = statusFunction;
+        server = ctx instanceof ServiceRequestContext;
```

Comment:
I was rather thinking of adding `boolean server` parameter to the constructor. There are two callers of this constructor and we know it's for server or client there.

Reply:
Agreed. If the type hierarchy is complex, `instanceof` has some cost.